src/analyzer/protocol/dns/DNS.cc

```diff
@@ -1372,8 +1372,8 @@ bool DNS_Interpreter::ParseRR_A(detail::DNS_MsgInfo* msg, const u_char*& data, i
 bool DNS_Interpreter::ParseRR_AAAA(detail::DNS_MsgInfo* msg, const u_char*& data, int& len, int rdlength) {
     uint32_t addr[4];
 
-    for ( int i = 0; i < 4; ++i ) {
-        addr[i] = htonl(ExtractLong(data, len));
+    for ( uint32_t& a : addr ) {
+        a = htonl(ExtractLong(data, len));
```

Comment:
Repeat from before - I'm quite unsure if for parsing code and fixed limits the `for ( uint32_t& a : addr )` is better - a mutable reference being assigned hides things for me :-( Anyone else has an opinion?